Hydration UI Setup [KIR-316]

lheld-pnmac opened the pull request:
These changes set up the basic chrome for the "New Job" view in Hydration to enable exploration of drag & drop ETL flow.

Work done:
* Updated list items and links in Hydration nav bar dropdown
* Created NewJobHeader component with breadcrumbs containing link to "View Jobs" view.
* Added center and right section placeholders that will eventually contain the canvas and toolbar, respectively.

Screenshot of view:
![image](https://user-images.githubusercontent.com/57911979/70098305-cc282380-15e8-11ea-9e73-d2f895ffbd13.png)

Jake-Lindvall commented on src/Components/Scenes/SideNavigation/HydrationListItem/HydrationListItem-Container.js:
```diff
@@ -4,8 +4,9 @@ import HydrationListItem from './HydrationListItem';
     hydrationListItemsName: [
-      { label: 'Add / Transform', link: '/hydration/add-transform' },
-      { label: 'Destinations', link: '/hydration/destinations' }
+      { label: 'New Destination', link: '/hydration/new-destination' },
```
Curious, are these URLs spec'd out anywhere? Legit just wondering

sfowles replied:
The only specs I'm aware of are these. Not very detailed though, and absolutely open to suggestions.
![image](https://user-images.githubusercontent.com/8948879/70161361-cd9d2e80-1670-11ea-8a6e-91de1d0fccd4.png)

sfowles replied:
Totally derped my last comment, sorry. Here's a mockup of the current spec for this part of the nav. The URLs aren't specified, but the labels are. https://www.figma.com/file/hhRwCxtuZb27X3XFI03rqZ/Kirby-Components?node-id=6%3A417

Nothing there has been tested yet though.

lheld-pnmac replied:
The URLs aren't specifically but I extrapolated from wording Scott's mocks :)

SelcukEdma commented:
looks good to me
Just small think,I know this is just to setup but It would be better if you follow our fileName convention
FeatureName Folder
- FeatureName-Container.js
- FeatureName.js

robertwollny-pnmac commented on src/Components/Scenes/Hydration/NewJob/NewJob.js:
```diff
@@ -0,0 +1,40 @@
+const chromeStyles = makeStyles(theme => ({
+  container: {
+    position: 'fixed',
+    zIndex: 1101, // Positions container on top of original header bar
```
Do you think it will be better to conditional render the appbar based on the pathname that way we don't have to use zIndex. Or we could have a hydration header component. What are your thoughts?

lheld-pnmac replied:
Yeah, I was super conflicted about this as well! I was trying to avoid having any messy conditional render logic within the higher-level PageWrapper component that was dependent on routing, but I definitely agree zIndex is gross haha. My rationale was that there's also going to be a good amount of interactivity with the header, see: [here](https://www.figma.com/file/jTZDpSj1DmDmwZ9o4IWGiI/Upload-Scripts?node-id=1%3A1198). Additionally, we need the canvas to fill the entirety of the page.

I'm thinking something in PageWrapper like: `{currentPath === '/hydration/new-job' ? null : <AppBarContainer />}` -- that way we don't have to rely on zIndex, but we're still able to have a custom header for the NewJob view. Lmk what you think!

robertwollny-pnmac replied:
That's exactly what I was thinking. 🔥

Jake-Lindvall commented on src/Components/Scenes/Hydration/NewJob/NewJob.js:
```diff
@@ -0,0 +1,40 @@
+    display: 'flex'
+  },
+  // Placeholder pattern until canvas is implemented
+  canvas: {
```
I hope you rolled your own 100% on the CSS functions here

lheld-pnmac replied:
Oh you know it

Jake-Lindvall commented on src/Components/Scenes/Hydration/NewJob/NewJob.js:
```diff
@@ -0,0 +1,40 @@
+      <Box className={classes.canvas}>
+        <NewJobHeader jobName="My New Job"></NewJobHeader>
+      </Box>
+      {/* Placeholder box until toolbar is implemented */}
```
Nitpicky thing on any placeholders (and I'm realizing I could do this in a couple places too) - at what level of severity do we want to add and track something like `TODO`s?

Almost more of an open question for the team. Thanks for adding comments to clarify in general!

sfowles replied:
On remote branches I think anything goes, but to keep the main codebase clean, I'm a big fan of not merging any placeholder content or incomplete functions. My experience is, things generally get forgotten about much more quickly once they've been merged, advancing codebase entropy. That's obv just my 2 cents!

lheld-pnmac replied:
I'd be happy to throw a bunch of TODO's at this to clarify what still needs to be fleshed out. I tend to feel that without the placeholders though, PRs will get too huge when trying to implement these components in one fell swoop. Since the Hydration UI is essentially one of our highest priorities and we will continue work on it immediately after, it doesn't bother me in this case.

lheld-pnmac replied:
Another note, this layout might also be changing once I start exploring with the project storm react-diagrams and how interactions between the toolbar and canvas will look. I figured this at least creates a place where I can continue with that spike work and see what will work best.

lheld-pnmac commented:
Hey guys! I updated with:
* Conditional render of main appbar
* Removed zIndex messiness
* Added down arrow next to Job Name breadcrumbs
* Added TODOs for future work
* Renamed files with Container structure convention
